### PR TITLE
fix(ruby): support ruby alpha format

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

Syntax like `3.0.0.alpha.0` was not supported by `getNextVersion`.

### CODE REVIEW

File: scripts/release/createReleasePR.ts [hunk 1]
```diff
@@ -179,7 +179,7 @@ export function getNextVersion(current: string, releaseType: semver.ReleaseType
   // python alpha releases have a pattern like X.Y.ZaN
   // where a means alpha and N can be any digit representing the alpha version
   // see https://peps.python.org/pep-0440/
```
Comment (Fluf22) [inline scripts/release/createReleasePR.ts:181]: Isn't this comment not related anymore?

**Review by Fluf22 — APPROVED** [review]

File: scripts/release/createReleasePR.ts (hunk 1 above)
Comment (shortcuts) [reply scripts/release/createReleasePR.ts:181]: It can be expanded to ruby indeed

Comment (millotp) [reply scripts/release/createReleasePR.ts:181]: it's still relevant for python, but also support ruby now, I can add a small com

**Review by shortcuts — COMMENTED** [review]
Sheeesh

File: scripts/release/createReleasePR.ts (hunk 1 above)
Comment (Fluf22) [reply scripts/release/createReleasePR.ts:181]: Or maybe remove it? I feel like the regex is self explanatory 😁 
It's true we could document every language specificity to know why the regex get more complex each time.

Comment (millotp) [reply scripts/release/createReleasePR.ts:181]: hopefully we won't stumble across new version format ! this should support everything

Comment (millotp) [reply scripts/release/createReleasePR.ts:181]: quoi que, ruby has `[2.0.0.pre.alpha.2](https://rubygems.org/gems/algolia/versions/2.0.0.pre.alpha.2)`

**Review by Fluf22 — APPROVED** [review]
Let's go

**Review by shortcuts — APPROVED** [review]
lovely

**Pull request merged**